Sources/AccessibilitySnapshot/Core/Swift/Classes/HitTargetSnapshotView.swift

```diff
@@ -126,15 +136,72 @@ public enum HitTargetSnapshotUtility {
 
                 // Finish the scan line if necessary.
                 if let lastHit = lastHit, let lastHitView = lastHit.1 {
-                    drawScanLine(
-                        for: lastHitView,
-                        startingAtX: lastHit.0,
-                        endingAtX: bounds.maxX,
+                    scanLine.append(((lastHit.0...bounds.maxX), lastHitView))
+                }
+
+                return scanLine
+            }
+
+            func drawScanLine(_ scanLine: ScanLine, y: CGFloat, lineHeight: CGFloat) {
+                for segment in scanLine {
+                    drawScanLineSegment(
+                        for: segment.view,
+                        startingAtX: segment.xRange.lowerBound,
+                        endingAtX: segment.xRange.upperBound,
                         y: y,
-                        lineHeight: pixelWidth
+                        lineHeight: lineHeight
                     )
                 }
             }
+
+            func scanLinesEqual(_ a: ScanLine, _ b: ScanLine) -> Bool {
+                return a.count == b.count
+                    && zip(a, b).allSatisfy { aSegment, bSegment in
+                        aSegment.xRange == bSegment.xRange && aSegment.view === bSegment.view
+                    }
+            }
+
+            // In some cases striding by 1/3 can result in the `to` value being included due to a floating point rouding
+            // error, in particular when dealing with bounds with a negative y origin. By striding to a value slightly
+            // less than the desired stop (small enough to be less than the density of any screen in the foreseeable
+            // future), we can avoid this rounding problem.
+            let stopEpsilon: CGFloat = 0.0001
```

Comment:
If we ever see screens with density high enough that this is an issue, I will eat my futuristic hat

Reply:
Maybe by the iPhone C